### PR TITLE
P5 play fix

### PR DESCRIPTION
**Pull request by islemaster**
Pulls a subtle `Animation.goToFrame()` fix from molleindustria/p5.play#111 into our own copy of p5.play, along with its tests.

### CODE REVIEW

File: apps/test/unit/gamelab/GameLabSpriteTest.js [hunk 1]
```diff
@@ -700,6 +700,174 @@ describe('GameLabSprite', function () {
...
+        // Play backwards (correctly this time)
+        animation.goToFrame(0);
+        animation.update();
+        expect(animation.getFrame()).to.equal(6);
+
+        // Try going to a positive frame out of bounds
+        // Unless ignored, we'd expect the animation to run forward again
+        animation.goToFrame(animation.images.length);
+        animation.update();
+        expect(animation.getFrame()).to.equal(5);
```
Comment (caleybrock) [inline apps/test/unit/gamelab/GameLabSpriteTest.js:855]: isn't the length > 6 why would we end up at 5 here?

File: apps/test/unit/gamelab/GameLabSpriteTest.js [hunk 2]
```diff
@@ -700,6 +700,174 @@ describe('GameLabSprite', function () {
...
+        expect(animation.playing).to.be.true;
+
+        animation.goToFrame(target);
+        expect(animation.playing).to.be.true;
+      }
+    });
+
+    it('plays the animation forward to the target frame when target > current', function () {
+      animation.changeFrame(1);
+      animation.goToFrame(4);
```
Comment (caleybrock) [inline apps/test/unit/gamelab/GameLabSpriteTest.js:750]: you're right this is super confusing and I would've guessed the expected behaviour totally wrong. Thanks for writing tests!

File: apps/test/unit/gamelab/GameLabSpriteTest.js (hunk 1 above)
Comment (islemaster) [reply apps/test/unit/gamelab/GameLabSpriteTest.js:855]: Just above here we call `animation.goToFrame(0)` when our current frame is 7 - this causes our animation to start playing backwards.  We do one update to verify that we're playing backwards, which puts the current frame at 6.

Then, we call `animation.goToFrame(animation.images.length)` where `length` here is 10, and valid frame indices are in the range 0-9, so it's an invalid parameter.  The expected behavior in this case is to do nothing, so we continue playing backwards - which is why we end up at frame 5.

Comment (caleybrock) [reply apps/test/unit/gamelab/GameLabSpriteTest.js:855]: ah okay that makes sense thank you!

**Pull request merged**